plonky2/src/plonk/circuit_builder.rs

```diff
@@ -343,6 +343,13 @@ impl<F: RichField + Extendable<D>, const D: usize> CircuitBuilder<F, D> {
         HashOutTarget::from_vec(self.add_virtual_targets(4))
     }
 
+    /// Registers a new `HashOutTarget` as a public input. `NUM_HASH_OUT_ELTS`
+    /// being hardcoded to 4, it internally adds 4 virtual targets in a
+    /// vector fashion.
+    pub fn add_virtual_hash_public_input(&mut self) -> HashOutTarget {
+        HashOutTarget::from_vec(self.add_virtual_public_input_arr::<4>().into())
```

Comment:
`HashOutTarget` stores elements internally as an array, and implements `From<[Target; NUM_HASH_OUT_ELTS]>`, so could just do:
```suggestion
        HashOutTarget::from(self.add_virtual_public_input_arr::<4>())
```
so that we don't need the allocation.